### PR TITLE
Tralalere

### PR DESCRIPTION
**Pull request by SebastienTainon**

### CODE REVIEW

File: frontend/stepper/index.ts [hunk 1]
```diff
@@ -728,74 +755,103 @@ function* stepperEnabledSaga(app: App) {
...
+            // if (runData.result || (!runData.result && runData.steps && runData.steps >= Codecast.runner._steps + 10)) {
+            //     newDelay = newDelay / 4;
+            // }
+            const t = Codecast.runner._steps / runData.steps;
+            const y0 = newDelay;
+            const y1 = newDelay / 40;
+            const y2 = newDelay / 40;
+            const y3 = newDelay;
+
+            newDelay = (1-t)*((1-t)*((1-t)*y0+t*y1)+t*((1-t)*y1+t*y2))+t*((1-t)*((1-t)*y1+t*y2)+t*((1-t)*y2+t*y3));
```
Comment (mblockelet) [inline frontend/stepper/index.ts:812]: Maybe a comment explaining the logic between the formula?

I understand it's the formula which accelerates and decelerates over time from newDelay to newDelay / 40, but a comment could help.

**Review by mblockelet — APPROVED** [review]
:+1:

File: frontend/stepper/index.ts (hunk 1 above)
Comment (SebastienTainon) [reply frontend/stepper/index.ts:817]: Ok, I've added a comment to explain :+1:

File: frontend/stepper/js/blockly_runner.ts [hunk 2]
```diff
@@ -87,8 +88,18 @@ export default class BlocklyRunner extends AbstractRunner {
 
     reportBlockValue(id, value, varName) {
         // Show a popup displaying the value of a block in step-by-step mode
-        console.log('report block value', id, value, varName, this.strings);
         if (this.context.display && this.stepMode) {
+            log.getLogger('blockly_runner').debug('report block value', {id, value, varName}, this.strings);
+
+            // Fix for Scratch because in ext/scratch/fixes.js, we report the value as varName = varValue. I don't know why, it should perhaps be changed.
```
[the remark this answers is not in the document]
Comment (SebastienTainon) [reply frontend/stepper/js/blockly_runner.ts]: Ok, in Codecast we always display varName= varValue anyway, so it should be good. I've updated my comment :+1:

**Pull request merged**